src/libraries/System.Private.CoreLib/src/System/Environment.SunOS.cs

```diff
@@ -8,6 +8,6 @@ namespace System
     public static partial class Environment
     {
         public static long WorkingSet =>
-            (long)(Interop.procfs.TryReadProcessStatusInfo(ProcessId, out Interop.procfs.ProcessStatusInfo status) ? status.ResidentSetSize : 0);
+            (long)(Interop.procfs.TryReadProcessStatusInfo(Interop.procfs.ProcPid.Self, out Interop.procfs.ProcessStatusInfo status) ? status.ResidentSetSize : 0);
```

Comment:
This breaks the build on illumos. Was this intentional?

Reply:
No. I assume the fix is straight forward?

Reply:
Yes, I will push it to my ongoing branch https://github.com/dotnet/runtime/pull/105178.

Reply:
Thanks!